src/day8/re_frame/trace.cljs

```diff
@@ -213,20 +214,32 @@
                         (if (< string-size-limit (count tag-str))
                           (str (subs tag-str 0 string-size-limit) " ...")
                           tag-str))]]))))))
+
 (defn render-trace-panel []
   (let [filter-input               (r/atom "")
         filter-items               (r/atom (localstorage/get "filter-items" []))
         filter-type                (r/atom :contains)
         input-error                (r/atom false)
+        categories                 (r/atom #{:event :sub/run :sub/create})
         trace-detail-expansions    (r/atom {:show-all? false :overrides {}})]
     (add-watch filter-items
                :update-localstorage
                (fn [_ _ _ new-state]
                  (localstorage/save! "filter-items" new-state)))
     (fn []
-      (let [showing-traces       (if (= @filter-items [])
-                                   @traces
-                                   (filter (apply every-pred (map query->fn @filter-items)) @traces))
+      (let [toggle-category-fn   (fn [category-keys]
+                                   (swap! categories #(set (if (empty? (set/intersection @categories category-keys))
+                                                             (concat category-keys @categories)
+                                                             (remove category-keys @categories)))))
```

Comment:
The function that you pass to `swap!` receives the current value of the atom as argument so you could use `%` instead of `@categories`.

Reply:
ahh true, thank you!